engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -287,26 +287,23 @@ def scc_activate_product
           auth = nil
           auth = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
           mode = nil
+          token_not_provided = !params.key?(:token) || params[:token].empty?
           if @system.byos?
             mode = 'byos'
           elsif !@product.free? && @product.extension?
-            regcode_error_message = 'A registration code is required to activate this product'
-            raise ActionController::TranslatedError.new(regcode_error_message) if (@system.payg? || @system.hybrid?) && params[:token].nil?
-
-            # system is not BYOS and
-            # product is a non free extension
-            # and a registration code is provided
-            mode = 'hybrid'
-            # the extensions must be the same version and arch
-            # than base product
-            base_prod = @system.products.find_by(product_type: :base)
-            if base_prod.present? && @product.arch == base_prod.arch && @product.version == base_prod.version
-              request.headers['proxy_byos_mode'] = mode
-              request.headers['scc_login'] = @system.login
-              request.headers['scc_password'] = @system.password
-              request.headers['instance_data'] = params[:instance_data]
-              request.headers['hwinfo'] = params[:hwinfo]
-              response = SccProxy.announce_system_scc(auth, request.headers)
+            unless token_not_provided
+              mode = 'hybrid'
+              # the extensions must be the same version and arch
+              # than base product
```

Comment:
So if the version + arch need to match, why is this combined with the condition that determines if an SCC announce is needed? 
How about: 
`raise 'Incompatible extension product' unless @product.arch == base_prod.arch && @product.version == base_prod.version`